sentry-core/src/main/java/io/sentry/core/Hub.java

```diff
@@ -27,12 +28,23 @@
   private volatile boolean isEnabled;
   private final @NotNull Deque<StackItem> stack = new LinkedBlockingDeque<>();
 
+  private static final AtomicBoolean ranDefaultIntegrations = new AtomicBoolean(false);
+
   public Hub(@NotNull SentryOptions options) {
     this(options, createRootStackItem(options));
 
-    // Register integrations against a root Hub
+    // Register integrations against the current hub at the given time
+    // Default integrations should run only once
+    if (ranDefaultIntegrations.getAndSet(true)) {
+      if (options.isDefaultIntegrationsEnabled()) {
+        for (Integration integration : options.getDefaultIntegrations()) {
+          integration.register(HubAdapter.getInstance(), options);
+        }
+      }
+    }
+
     for (Integration integration : options.getIntegrations()) {
-      integration.register(this, options);
+      integration.register(HubAdapter.getInstance(), options);
     }
   }
```

Comment:
@bruno-garcia @HazAT right now it won't `defaultIntegrations.register` twice, is it the correct behavior or this should only be done for integrations that are global hooks? by coincidence, right now all the `defaultIntegrations` are global hooks, so it makes sense to run only once, but if we add a new `defaultIntegration` that is not a global hook, it won't be able to register again.